libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1721,8 +1725,99 @@ void IRGeneratorForStatements::appendExternalFunctionCall(
 
 	templ("forwardingRevert", m_utils.forwardingRevertFunction());
 
-	solUnimplementedAssert(funKind != FunctionType::Kind::RIPEMD160, "");
-	solUnimplementedAssert(funKind != FunctionType::Kind::ECRecover, "");
+	m_code << templ.render();
+}
+
+void IRGeneratorForStatements::appendBareCall(
+	FunctionCall const& _functionCall,
+	vector<ASTPointer<Expression const>> const& _arguments
+)
+{
+	FunctionType const& funType = dynamic_cast<FunctionType const&>(type(_functionCall.expression()));
+	solAssert(
+		!funType.bound() &&
+		!funType.takesArbitraryParameters() &&
+		_arguments.size() == 1 &&
+		funType.parameterTypes().size() == 1, ""
+	);
+	FunctionType::Kind const funKind = funType.kind();
+
+	solAssert(funKind != FunctionType::Kind::BareStaticCall || m_context.evmVersion().hasStaticCall(), "");
+	solAssert(funKind != FunctionType::Kind::BareCallCode, "Callcode has been removed.");
+	solAssert(
+		funKind == FunctionType::Kind::BareCall ||
+		funKind == FunctionType::Kind::BareDelegateCall ||
+		funKind == FunctionType::Kind::BareStaticCall, ""
+	);
+
+	solAssert(!_functionCall.annotation().tryCall, "");
+	Whiskers templ(R"(
+		// temporarily store the free memory pointer
+		let <freeMemoryStored> := mload(<freeMemoryPointer>)
```

Comment:
This is only needed for the ``needsEncoding`` case, isn't it? If so, then resetting as well...

Reply:
I can pull both inside the check.

Reply:
Actually I'm wondering whether `convert` would not be better suited than encodePacked. It has the benefit that it does nothing on bytes memory and string memory, but would simplify the template.

Reply:
Oh, sorry, this is still left to do.

Reply:
Actually this is all unneccessary - the tuple encoder does not modify the free memory pointer.